src/main/java/edu/gonzaga/Hand.java

```diff
@@ -1,32 +1,186 @@
+     * @author Izzy T
+     */
+    public ArrayList<BasicCard> getBasicCardsInHand() {
+        ArrayList<BasicCard> basicArray = new ArrayList<BasicCard>();
+        for (Integer i = 0; i < cardList.size(); i++) {
+            if (cardList.get(i).getCardType().equals("BasicCard")) {
+                BasicCard basicCard = (BasicCard) cardList.get(i); // type casting card to basic card type so I can add it to array for Basic Cards
+                basicArray.add(basicCard);
+            }
+        }
+        return basicArray;
     }
 
-    public void checkCanCure() {
 
+    /**
+     * Removes the first instance of a card from a hand (if it exists)
+     * PASSES TESTS
+     * @param card the card that the player wants to remove from their hand - using generic type so that it works for basic and event cards
+     * @author Izzy T
+     */
+    public void discard(Card card) {
+        if (!cardList.contains(card)) {
+            System.out.println("This card is not in your hand");
+        }
+        else {
+            System.out.println("The card " + card.getCardType() + " was discarded.");
+            cardList.remove(card);
+        }
+    }
+
+    /**
+     * Checks if there are five cards of the desired color in a players hand - aka # of cards 
+     * PASSES TESTS
+     * @param cardColor the color of the city cards that we want to check for 
+     * @author Izzy T
+     */
+    public boolean checkCanCure(Color cardColor) {
+        Integer numCardsOfColor = 0;
+        for (Integer i = 0; i < cardList.size(); i++) {
+            if (cardList.get(i).getCardType().equals("BasicCard")) { 
+                BasicCard basicCard = (BasicCard) cardList.get(i); // type casting card to basic card type so I can access its Color
+                // then, if that card's color is the same as the color of interest, increment numCardsOfColor
+                if (basicCard.getColor() == cardColor) {
+                    numCardsOfColor++;
+                }
+            }
+        }
+        if (numCardsOfColor >= numCardsToCure) {
+            return true;
+        }
+        return false;
     }
 
-    public void getCardList() {
+    /**
+     * Accessor for the private member variable, cardList, in Hand 
+     * 
+     * @return the list of cards in a Hand, of type ArrayList<Card>
+     * @author Izzy T
+     */
+    public ArrayList<Card> getCardList() {
+        return this.cardList;
+    }
 
+    /**
+     * Restricts number of cards in a Hand - if card # exceeds max num allowed, then it requests one card be discarded until doesHandFollowLimit() is true
+     * 
+     * @author Izzy T
+     */
+    public void makeHandValid(Scanner userInput) {
+        int targetVal; // index of card list to delete a card from 
+        while (!this.doesHandFollowLimit()) {
+            System.out.println("Your hand exceeds the maximum card amount of " + handLimit + ". What card would you like to get rid of? ");    
```

Comment:
Can you add a //TODO: comment for adding functionality with the UI? We should probably also add a card to Trello